cms/views/cms_views.py

```diff
@@ -61,7 +61,15 @@ def get_featured_languages(self):
     def get_display_name(self, locale):
         language_code, _, country_code = locale.partition('_')
         term_code = languages.get(bibliographic=language_code).terminology
-        return Locale.parse(term_code).language_name
+
+        available_languages = dict(literal_eval(
+            (self.settings.get('available_languages', '[]'))))
+
+        try:
+            return Locale.parse(term_code).language_name
+        except UnknownLocaleError:
+            # Fallback value is the generated value in English or the code
```

Comment:
That's the problem, our libs don't recognise it so we can't have a localised version of the name.